docs/reference/mapping.asciidoc

```diff
@@ -36,6 +36,9 @@ treated. Examples of meta-fields include the document's
 A mapping type contains a list of fields or `properties` pertinent to the
 document.
 
+`index.mapping.field.name.length.limit`::
```

Comment:
Maybe this can be moved to the "Settings to prevent mappings explosion" section since there are already some mapping related settings there, although this is not really a setting that limits the number of fields (and thus is not really that relevant to prevent mapping explosion). I just think while this setting also fits here, its probably not really important enough to appear in this very general section.

If you agree its a good idea to move it I'd put it at the bottom of the refered section and add a short sentence that this setting isn't really something that adresses mappings explosion but might still be useful if you want to limit the field length. I would also state that it usually shouldn't be necessary to set it and the defaults are okay unless users start to add really huge numbers of fields with really long names (in the thousands probably))